django_celery_token_bucket/decorators.py

```diff
@@ -1,15 +1,24 @@
+from django.conf import settings
 from functools import wraps
 from queue import Empty
+from django_celery_token_bucket.bucket import TokenBucket
 
 
-def rate_limit(token_bucket: str, retry_backoff: int = 60):
+def rate_limit(token_bucket_name: str, retry_backoff: int = 60):
     def decorator_func(func):
         @wraps(func)
         def function(self, *args, **kwargs):
             with self.app.connection_for_read() as conn:
-                with conn.SimpleQueue(token_bucket, no_ack=True) as queue:
+                queue_name = f"{TokenBucket.QUEUE_PREFIX}{token_bucket_name}"
```

Comment:
This could be done after getting the `TokenBucket` from `settings.CELERY_TOKEN_BUCKETS` by using `TokenBucket.get_queue_name()`